application/src/main/java/org/togetherjava/tjbot/commands/moderation/WarnCommand.java

```diff
@@ -0,0 +1,134 @@
+            @NotNull String reason, @NotNull Guild guild, @NotNull SlashCommandEvent event) {
+        return event.getJDA()
+            .openPrivateChannelById(target.getId())
+            .flatMap(channel -> channel.sendMessage(
+                    """
+                            Hey there, sorry to tell you but unfortunately you have been warned in the server %s.
+                            If you think this was a mistake, please contact a moderator or admin of the server.
+                            The reason for the warning is: %s
+                            """
+                        .formatted(guild.getName(), reason)))
+            .mapToResult()
+            .map(Result::isSuccess);
+    }
+
+    private void warnUser(@NotNull User target, @NotNull Member author, @NotNull String reason,
+            @NotNull Guild guild) {
+        logger.info("'{}' ({}) warned the user '{}' ({}) for reason '{}'.",
+                author.getUser().getAsTag(), author.getId(), target.getAsTag(), target.getId(),
+                reason);
+
+        actionsStore.addAction(guild.getIdLong(), author.getIdLong(), target.getIdLong(),
+                ModerationUtils.Action.WARN, null, reason);
+
+    }
+
+    private static @NotNull MessageEmbed sendFeedback(boolean hasSentDm, @NotNull User target,
+            @NotNull Member author, @NotNull String reason) {
+        String dmNoticeText = "";
+        if (!hasSentDm) {
+            dmNoticeText = "(Unable to send them a DM.)";
+        }
+        return ModerationUtils.createActionResponse(author.getUser(), ModerationUtils.Action.WARN,
+                target, dmNoticeText, reason);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(USER_OPTION), "The target is null");
+        User target = targetOption.getAsUser();
+        Member author = Objects.requireNonNull(event.getMember(), "The author is null");
+        Guild guild = Objects.requireNonNull(event.getGuild(), "The guild is null");
+        String reason = Objects.requireNonNull(event.getOption(REASON_OPTION), "The reason is null")
+            .getAsString();
+
+        if (!handleChecks(guild.getSelfMember(), author, targetOption.getAsMember(), reason,
+                event)) {
+            return;
+        }
+
+        warnUserFlow(target, author, reason, guild, event).queue();
+    }
+
+    @SuppressWarnings("BooleanMethodNameMustStartWithQuestion")
+    private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
+            @Nullable Member target, String reason, @NotNull SlashCommandEvent event) {
+        if (target == null) {
+            if (!ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event)) {
+                return false;
+            }
+        } else {
+            // Check if the author can warn a user with higher perms.
```

Comment:
This comment is obsolete. It is pretty much identical with the method name. Its as if you write
```java
// Gets the name of the person
String name = person.getName();
```
Thats not how comments should be used.